resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -198,6 +198,7 @@ private[spark] object KubernetesConf {
       .replaceAll("\\.", "-")
       .replaceAll("[^a-z0-9\\-]", "")
       .replaceAll("-+", "-")
+      .replaceAll("^-", "")
```

Comment:
Is this still needed then, if you're simply validating and replacing the name if it starts with "-"?

Reply:
> Is this still needed then, if you're simply validating and replacing the name if it starts with "-"?

The  '^-' is used for all kubernetes resources. If the resource name  starts with "-",we replace it with `""` ,which can run normally

Reply:
?

Reply:
If we need to match the fabric8 regex bellow, shouldn't this be `.replaceAll("^[^a-z]", "")` to replace any leading/first non alpha character?